core/server/master/src/main/java/alluxio/master/file/meta/MountTable.java

```diff
@@ -149,7 +149,7 @@ public void remove() {
    */
   public void add(AlluxioURI alluxioUri, AlluxioURI ufsUri, long mountId, MountOptions options)
       throws FileAlreadyExistsException, InvalidPathException {
-    String alluxioPath = alluxioUri.getPath();
+    String alluxioPath = alluxioUri.getPath().isEmpty() ? "/" : alluxioUri.getPath();
```

Comment:
can you change over `alluxio.AlluxioURI#getPath` to return `"/"` or `mUri.getPath()`?
Or even create `mUri` in `alluxio.AlluxioURI` with this in mind?
then also modify the description of javadoc of `alluxio.AlluxioURI` and saying `scheme://host:123` is a valid path?
Also it will be nice to add unit test in `AlluxioUriTest`

Reply:
That's the first option I considered and tried, but it breaks many other unit tests.
For example, the `AddressUtil.replaceHostName` will append `/` if it's omitted.
Think it twice, I think AlluxioURI should retain the original information, but not processed.

Reply:
ok, I am fine with the current workaround.
can you create a JIRA tracking this issue and put a TODO here stating why we have this condition ?